docs/source/tutorials/regression.ipynb

```diff
@@ -0,0 +1,521 @@
+      "text/plain": [
+       "<Figure size 1400x500 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# Generate true errors\n",
+    "true_errors_index = np.where(data[\"grade\"] != data[\"true_grade\"])[0]\n",
+    "plot_data(\n",
+    "    data_x=data[\"exam_3\"],\n",
+    "    data_y=data[\"grade\"],\n",
+    "    circles=true_errors_index,\n",
+    "    title=\"Messy Regression dataset\",\n",
+    "    xlabel=\"exam_3 feature\",\n",
+    "    ylabel=\"grade (Y value)\",\n",
+    ")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In the above plot, `grade (Y value)` is plotted against one of the features in the dataset (`exam_3`). We have circled the examples that were considered as `true_error` in **Red** . "
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will use sklearn API `XGBRegressor` from `xgboost` as the regressor for this tutorial. `xgboost` provides easy to use interface to process categorical variables. In order to make inputs compatible with `xgboost`, we need to divide data in `X` and labels `y`. This is demonstrated in the code below: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# xgboost automatically factors categorical variable, you just need to mark the columns as category\n",
+    "data.notes = data.notes.astype(\"category\")\n",
+    "\n",
+    "# xgboost takes data and label seperately, so you will need to divide data accordingly.\n",
+    "X = data.drop([\"grade\", \"true_grade\"], axis=1)\n",
+    "y = data[\"grade\"]"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's start with initializing the model with relevant parameters. As mentioned earlier we are using `xgboost` for this tutorial. To handle categorical variables, we specifically need to set `enable_categorical` flag to `True`. Note that, support for the categorical variable is in the experimental stage and doesn't support the auto-selection of `tree_method`. Therefore, you will need to specify `tree_method` from supported types. More details can be found [here](https://xgboost.readthedocs.io/en/stable/tutorials/categorical.html)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "SEED = 1\n",
+    "num_crossval_folds = 5\n",
+    "\n",
+    "model = XGBRegressor(\n",
+    "    tree_method= \"hist\",\n",
+    "    n_estimators = 100,  \n",
+    "    enable_categorical = True, \n",
+    "    random_state = SEED)\n",
+    "\n",
+    "# get predictions using cross-validation\n",
+    "predictions = cross_val_predict(\n",
+    "    estimator=model, X=X, y=y, cv=num_crossval_folds, method = \"predict\"\n",
+    ")"
+   ]
+  },
```

Comment:
Let's add another cell after this that demonstrates how to evaluate the R2 of the out-of-sample predictions from cross-validation.

This cell should be structured similar to our text tutorial:  https://docs.cleanlab.ai/stable/tutorials/text.html

in that notebook, search for: "An additional benefit of cross-validation is that it facilitates more reliable evaluation of our model than a single training/validation split."